### ISSUE TITLE
Named parameters for List/Array/Nel/Nea concat

### ISSUE THREAD

**Issue by andywhite37** [reazen/relude#150]
The `concat` function for list/array/Nel/Nea should probably have at least one named parameter, because it's really easy to screw up the order of arguments, especially when used with `|>`.

I'm not sure what the best names should be - `lhs` and `rhs`?  Only name one of them?

@mlms13 @johnhaley81 - any thoughts?

```reason
let concat = (~lhs: list('a), rhs: list('a)): list('a) => ...`

["right", "values"] |> concat(~lhs=["left", "values"]) // ["left", "values", "right", "values"]
```

Or maybe we don't want named args at all for `concat`?

**Comment by johnhaley81** [comment]
Some more name ideas:

`prefix`/`suffix`
`with`/`on`

**Comment by andywhite37** [comment]
Yeah, I think I like `prefix`/`suffix`

**Comment by andywhite37** [comment]
Added `concatNamed` to `SemigroupExtensions` and `SemigroupAnyExtensions`, which affects `Array`, `List` and `NonEmpty.*`.

Released in 0.34.0
